app/main.py

```diff
@@ -2,5 +2,18 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    our_dict = {}
+
+    def wrapper(*args) -> int:
+
+        if args not in our_dict:
```

Comment:
Using `args` as a dictionary key assumes that all arguments are hashable. If any of the arguments are mutable (like lists), this will raise a `TypeError`. Consider ensuring that only hashable types are used as arguments or converting mutable types to immutable ones (e.g., using tuples).